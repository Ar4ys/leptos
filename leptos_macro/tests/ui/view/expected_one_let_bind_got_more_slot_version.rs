use leptos::*;

#[slot]
struct Then<C, IV>
where
    C: Fn(String) -> IV,
    IV: IntoView,
{
    children: C,
}

#[component]
fn Slot<C, IV>(then: Then<C, IV>) -> impl IntoView
where
    C: Fn(String) -> IV,
    IV: IntoView,
{
    _ = then;
}

fn main() {
    view! {
        <Slot>
            <Then
                slot
                let:item
                let:extra_item
            >
                <p>{item}</p>
            </Then>
        </Slot>
    };
}
