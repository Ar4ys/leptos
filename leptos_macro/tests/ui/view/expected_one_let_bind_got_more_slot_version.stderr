error[E0593]: closure is expected to take 1 argument, but it takes 2 arguments
  --> tests/ui/view/expected_one_let_bind_got_more_slot_version.rs:29:17
   |
29 |                 <p>{item}</p>
   |                 ^^^^^^^^^^^^^
   |                 |
   |                 expected closure that takes 1 argument
   |                 takes 2 arguments
